MoreLinq.Test/TakeLastTest.cs

```diff
@@ -57,5 +63,27 @@ public void TakeLastDisposesSequenceEnumerator()
                 seq.TakeLast(1).Consume();
             }
         }
+
+        [Test]
+        public void TakeLastOptimizedForCollections()
+        {
+            var collection = new OnceEnumerationCollection<int>(Enumerable.Range(1, 10));
```

Comment:
We already have `UnenumerableList<>` that can be used here to test the same thing:

```c#
var collection = new UnenumerableList<int> { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
```

`OnceEnumerationCollection<>` should be removed.